### PR TITLE
Add the Vasicek1F.cs as an additional short-rate model.

### LINKED ISSUE #31: Implement Vasicek1F as an Alternative Short-Rate Model
**Context**
We currently use the HullWhite1F model for short rate simulation. It’s great for fitting the yield curve, but it’s a bit complex due to the time-dependent drift. Let’s add the Vasicek1F model as a simpler alternative.

**Details**
- Same mean-reverting idea, but with constant drift.
- No curve fitting needed.
- Easier to test and compare.
- Useful for benchmarking and educational purposes.
- Plan is to create a Vasicek1F class that implements NumeraireSimulator, with similar structure to HullWhite1F.

### PR DESCRIPTION
**Pull request by rwx-oberholzer-marcelle**
✅ # Pull Request Checklist

### General
* [x] Unless this is an intermediate review, PR is not a draft.

### 🔍 Self-Review
* [x] I have performed a self-review and passed it.

### 👥 Collaboration
* [x] Reviewers assigned and feedback requested.
* [x] All review comments have been addressed or responded to.

### General
* [ ] Destination branch is correct.
* [ ] Code is scoped to the PR’s purpose.
* [ ] No debugging code, print statements, or unrelated changes.
* [ ] Code is clean, well-commented, and follows project standards.
* [ ] Documentation is updated, if applicable.

### 🧪 Testing & Build
* [ ] All relevant tests (unit/integration) pass locally.
* [ ] Edge cases and error handling are considered.
* [ ] No new warnings or errors in the build or test output.

### CODE REVIEW

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        public override void Reset()
+        {
+            _allDates = new List<Date>();
+        }
+
+        public override void SetRequiredDates(MarketObservable index, List<Date> requiredDates)
+        {
+            if (_allDates == null) _allDates = requiredDates;
+            else _allDates.AddRange(requiredDates);
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs]: Do other models do this? I can see the results being a bit confusing for someone if _allDates already has some values.

I say this since the name starts with "Set", as opposed to "Add".

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        public override void SetNumeraireDates(List<Date> requiredDates)
+        {
+            if (_allDates == null) _allDates = requiredDates;
+            else _allDates.AddRange(requiredDates);
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs]: Do other models do this? I can see the results being a bit confusing for someone if _allDates already has some values.

I say this since the name starts with "Set", as opposed to "Add".

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        public override void Prepare(Date anchorDate)
+        {
+            _anchorDate = anchorDate;
+            _allDates.Insert(0, anchorDate);
+            _allDates = _allDates.Distinct().ToList();
+            _allDates.Sort();
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs]: This shows how you need to look at specifying the right types to use. Here _allDates is a List<Date>, but you actually requires distinct elements, and you want them sorted in ascending order.

Do some research to see if there's a C# Collection that supports one or both of these requirements and chat to me about it.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]
Also add a test that validates the model, ideally with results from an external source/tool.

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs (hunk 1 above)
Comment (rwx-oberholzer-marcelle) [reply QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs]: @robertjacquier-rw , this is what they did for the HullWhite1F model. I understand the confusion, but to match the base class override (NumeraireSimulator), I added xml documentation to clarify that the method appends to the list rather than replacing it. I can just change the name to AddRequiredDates if you think that is a better option?

**Comment by rwx-oberholzer-marcelle** [comment]
> Also add a test that validates the model, ideally with results from an external source/tool.

@robertjacquier-rw, I added a unit test that compares the bond price from our simulation to a known formula. (I used the closed-form Vasicek bond pricing formula that's implemented in external tools like QuantLib).

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs [hunk 4]
```diff
@@ -21,7 +21,8 @@ public class Vasicek1F : NumeraireSimulator
         private readonly Currency _currency;
 
         [JsonIgnore] private Date _anchorDate;
-        [JsonIgnore] private List<Date> _allDates;
+        //[JsonIgnore] private List<Date> _allDates;
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs]: No need to comment code like this and commit it, remember we have source control :) Just delete the line.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            _r0 = r0;
+            _currency = currency;
+            _floatRateIndices = floatRateIndices != null ? floatRateIndices.ToList() : new List<FloatRateIndex>();
+        }
+
+        public override void Reset()
+        {
+            _allDates = new SortedSet<Date>();
+        }
+        /// <summary>
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs:45]: Spacing.

File: QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        }
+
+        public override void Prepare(Date anchorDate)
+        {
+            _anchorDate = anchorDate;
+            _allDates.Add(anchorDate);
+
+            _allDatesDouble = _allDates.Select(d => (double)d).ToArray();
+            _dist = new NormalDistribution();
+            Generator.Seed = -1585814591;
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Valuation/Models/Rates/Vasicek1F.cs:70]: Store the magic number in a const near the top of the file with a good name. This makes it easier to spot as the class is viewed, and easier to maintain.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]
